crates/swc_ecma_lints/src/rules/no_alert.rs

```diff
@@ -182,4 +206,28 @@ impl Visit for NoAlert {
             expr.visit_children_with(self);
         }
     }
+
+    fn visit_class(&mut self, class: &Class) {
+        self.inside_class = true;
```

Comment:
I think this will error if it's nested, as after calling `visit_class` for a nested class, it will be `false`.

Reply:
You can use `let old_inside_class = self.inside_class`.

Reply:
You mean this case? 

```
class X {
  method() {
    class Y () {}
  }
}
```

Reply:
Not exactly. I meant 

```js
class X {
  method() {
    
  }
  
  method2() {
    class Y () {}
    this.alert()
  }
}

```

but your example may also break this.